app/js/index.js

```diff
@@ -48,23 +50,30 @@ Or we could just put the main logic back in index.html, even though that's not a
 //initialization of the entire map
 
 function init() {
-  // don't initialize until user has uploaded a .csv file
-  if (document.getElementById('input_csv').files[0] == null) {
-    console.log("Cartograms 4 All: Waiting for user inputted CSV file");
-    return;
+  // Start with default data and topo for user
+  // Switch to user data when given
+  if (userSessionCookie == null) {
+    userSessionCookie = readCookie('userSessionCookie');
   }
-  // CODE TO TEST FUNCTIONALITY OF writeToServer() and readFromServer()
-  SESSION_ID = readCookie('userSessionCookie');
 
-  //var send_text = "my_text_to_save";
-  //writeToServer(SESSION_ID, send_text);
-  //console.log(SESSION_ID);
+  if (document.getElementById('input_csv').files[0] == null) {
+    userData = DEFAULT_DATA;
+  } else {
+      //File object is immutable, so it does not rename to make it unique per user in js
+      var csv = document.getElementById('input_csv').files[0];
+     //Save user input if it is given and override the default
+      if (csv != null) { 
+        saveCSV(csv); 
+        //userData = USER_DIRECTORY + csv.name;
+      } else {
+        //Avoid null user file
+        userData = DEFAULT_DATA;
+      }
+      //Add local file usage to avoid async js calls that breaks map
```

Comment:
I will have to add a small check inside our init function when we import other user's data over. I think it will be more appropriate to do in your branch.